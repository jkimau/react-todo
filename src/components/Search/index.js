import React from 'react';
import styled from 'styled-components';

const Input = styled.input`
  width: 200px;
  height: 25px;
  padding: 2px 10px;
  border-radius: 15px;
  border: 1px solid #ddd;
  line-height: 1;

  &:focus {
    border: none;
  }
`;

export default () => {
  return <Input type="text" placeholder="Please enter keywordss" />
};
