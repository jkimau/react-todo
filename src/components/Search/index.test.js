import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Search from '../Search';

describe('<Search/> component', () => {
  const compileComponent = () => {
    return shallow(<Search/>);
  }

  test('snapshot', () => {
    const tree = renderer.create(
      <Search/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should render input elelment', () => {
    const component = compileComponent();

    expect(component.dive().find('input').length).toBe(1);
    expect(component.prop('type')).toEqual('text');
    expect(component.prop('placeholder')).toEqual('Please enter keywords');
  });
});
