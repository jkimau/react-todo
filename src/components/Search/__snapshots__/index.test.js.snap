exports[`<Search/> component snapshot 1`] = `
<input
  className="kTPKMA"
  placeholder="Please enter keywords"
  type="text" />
`;
